lib/ansible/modules/find.py

```diff
@@ -154,6 +154,15 @@
             - When doing a C(contains) search, determine the encoding of the files to be searched.
         type: str
         version_added: "2.17"
+    max_matches:
+        description:
+            - Set the maximum number of matches to make before returning.
+            - Matches are made from the top, down (i.e. shallowest directory first).
+            - Set to V(-1) for unlimited matches.
+            - Default is unlimited matches.
+        type: int
+        default: -1
```

Comment:
use `None` for 'unset/unlimited'

Reply:
Will change. Am I correct in thinking that the type should still be `int`, given there's no `Optional[int]` type (https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#argument-spec)?

Reply:
Have changed. Looks like type should be `int` and `default` should be left to default to `None`.

Reply:
you can just remove it, `None` is the default for `default`. All types allow `None` as a default as it is 'unset'.

Reply:
Thanks - changed. Tests passing again.